source/manual/alerts/rabbitmq-idle-consumers.html.md

```diff
@@ -0,0 +1,78 @@
+---
+owner_slack: "#govuk-2ndline"
+title: 'RabbitMQ: Consumers idle'
+parent: "/manual.html"
+layout: manual_layout
+section: Icinga alerts
+last_reviewed_on: 2020-08-24
+review_in: 6 months
+---
+
+This checks whether the RabbitMQ consumers for a queue have been active in the
+last 5 minutes.
+
+Publishing API [sends a heartbeat message][heartbeat_messages] every minute to
+the following queues - `email-alert-service`, `cache_clearing_service-high` and
+`content_data_api`. This should ensure that the consumers are never idle
+for these queues.
+
+## How this check works
+
+Icinga connects to RabbitMQ's admin API to check on the activity of the
+consumers of a given RabbitMQ message queue. This script for this check lives
+in [puppet][idle-puppet-config].
+
+## Troubleshooting steps
+
+### Check the RabbitMQ logs
+
+* Check the logs for the `rabbitmq` application which can be achieved using
+  [this Kibana query][kibana-query]. Is there evidence of any errors?
+
+### Check the RabbitmQ Grafana dashboard
+
+This [Grafana dashboard][rabbitmq_grafana_dashboard] shows activity across
+multiple exchanges and queues. The main exchange we expect to be monitoring is
+`published_documents` which handles broadcasting to services such as search and
+email-alert-service when content changes across GOV.UK.
+
+Looking at the queue graphs we should look out for the following:
+
+High / growing amount of 'ready' messages, this indicates these messages are
+waiting to be processed by the consumer (e.g email-alert-service).
+
+We should also look out for a high amount of unacknowledged messages which
+implies that messages have been read by the consumer but the consumer has never
+sent back an ACK to the RabbitMQ broker to say that it has finished processing.
+If this is down to the consumer dying (its channel is closed, connection is
+closed, or TCP connection is lost) then RabbitMQ should `redeliver` the
+message, if we're seeing high redelivery rates this could indicate an issue
+with the consumer.
+
+To troubleshoot these symptoms:
+
+* Check the status of the consumer, e.g for email-alert-service look on the
+  email_alert_api machines - `sudo service email-alert-service status`
+* If the services are running we should look in the consumers application logs.
+* Check messages received by the consumer are actually being actioned (e.g
+  email-alert-api is receiving the messages email-alert-service has received
+  from RabbitMQ). To do this we could compare the content changes in
+  [email-alert-api][email-alert-api] against whats recorded in the
+  email-alert-service logs.
+
+### Check the RabbitMQ control panel
```

Comment:
I think we already know this from the first section - it's in the Grafana dashboard. While we may see _something_ by looking at the control panel, we can always write about that when we see this alert for real and know what it is specifically.

Can we remove this section?